### ISSUE TITLE
Covariant Protocol with get-only properties  are required to be invariant

### ISSUE THREAD

**Issue by MHDante** [microsoft/pyright#3331]
**Describe the bug**
Covariant Get-Only properties on interface are required to be invariant

**To Reproduce**
Copy code below. 
Code produces following error:

```
error: Type variable "T" used in generic protocol "IEnumerator" should be invariant (reportInvalidTypeVarUse)
```

**Expected behavior**
Code below should produce no error. 

That said, I'm not sure if this is due to my misunderstanding of covariance in python. But wrapping a method that returns a typevar in a `@property` decorator causes pyright to complain about the variable required to be invariant. This might be due to the decorator exposing a setter even if it is not set?

**Screenshots or Code**

```python

# this complains
T = TypeVar("T", covariant=True)
class IEnumerator(Protocol[T]):
    @property
    def current(self) -> T: ...

# this works fine
T = TypeVar("T", covariant=True)
class IEnumerator(Protocol[T]):
    def current(self) -> T: ...

```

![image](https://user-images.githubusercontent.com/4652122/162760851-314c8f03-9c3c-4d60-b2b1-b9afd1997b6b.png)
![image](https://user-images.githubusercontent.com/4652122/162761184-4ff2e8b0-45c4-4344-a319-fe26ce2b8c92.png)

**VS Code extension or command-line**

pyright 1.1.236
![image](https://user-images.githubusercontent.com/4652122/162659643-462fff8c-51e8-41b6-87dd-5398f087c726.png)

![image](https://user-images.githubusercontent.com/4652122/162659700-9a9dd4d0-1722-4fdd-a4ad-407f5375454c.png)
pylance v2022.4.0

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.237, which I just published. It will also be included in the next release of pylance.
